src/test/ui/traits/static-method-generic-inference.stderr

```diff
@@ -9,8 +9,8 @@ LL |     let _f: base::Foo = base::HasNew::new();
    |
 help: use the fully-qualified path to the only available implementation
    |
-LL |     let _f: base::Foo = base::<::base::Foo as HasNew>::new();
-   |                               +++++++++++++++       +
+LL |     let _f: base::Foo = base::<Foo as HasNew>::new();
```

Comment:
that error was already wrong :sweat_smile: i guess the issue is that we shouldn't use the `trait_path_segment` but instead the whole path of the function except for the final segment :sparkles: